### PR TITLE
Updating antenna positions in UVData

### PR DESCRIPTION
**Pull request by kartographer**
Adds a method for updating antenna positions within a `UVData` object.

## Description

A new method called `update_antenna_positions` has been added to `UVData`, which allows a used to provide a dictionary of updates to antenna positions, and correspondingly update the uvw-positions of individual baselines as well as the visibility phases. Some additional changes/streamling of the `phase` and `_apply_w_proj` methods was also done, mostly to help make the code a bit more compact and eliminate some mostly-redundant code.

## Motivation and Context

This is a new feature, but one that's been long requested by folks within the SMA crowd, and may be useful for other users of reconfigurable arrays (e.g., CARMA, VLA, ALMA). Because antenna positions for such antennas are constantly being changed, it's not uncommon for improved solutions to be derived _after_ the time of observation. This method basically offers an easy, convenient way for implementing those changes, complementing the existing `phase` method (which adjusts astrometry/phasing position rather than antenna positions).

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

New feature checklist:
- [x] I have added or updated the docstrings associated with my feature using the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] I have updated the tutorial to highlight my new feature (if appropriate).
- [x] I have added tests to cover my new feature.
- [x] All new and existing tests pass.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/CHANGELOG.md).

### CODE REVIEW

File: pyuvdata/uvdata/uvdata.py [hunk 1]
```diff
@@ -5315,14 +5315,13 @@ def _apply_w_proj(self, new_w_vals, old_w_vals, select_mask=None):
         new_w_vals: float or ndarray of float
             New w-coordinates for the baselines, in units of meters. Can either be a
             solitary float (helpful for unphasing data, where new_w_vals can be set to
-            0.0) or an array of shape (Nselect,) (which is Nblts if select_mask=None).
+            0.0) or an array of shape (Nblts,).
         old_w_vals: float or ndarray of float
             Old w-coordinates for the baselines, in units of meters. Can either be a
             solitary float (helpful for unphasing data, where new_w_vals can be set to
```
Comment (bhazelton) [inline pyuvdata/uvdata/uvdata.py:5321]: The parenthetical here is currently identical to the "new_w_vals" docstring, which is a little confusing. Should this read "helpful for phasing unprojected data where old_w_vals can be set to 0.0"?

File: pyuvdata/uvdata/uvdata.py [hunk 2]
```diff
@@ -5315,14 +5315,13 @@ def _apply_w_proj(self, new_w_vals, old_w_vals, select_mask=None):
...
-            0.0) or an array of shape (Nselect,) (which is Nblts if select_mask=None).
+            0.0) or an array of shape (Nblts,).
         select_mask: ndarray of bool
-            Array is of shape (Nblts,), where the sum of all enties marked True is
-            equal to Nselect (mentioned above).
+            Array is of shape (Nblts,), which identifies which records to change.
```
Comment (bhazelton) [inline pyuvdata/uvdata/uvdata.py:5324]: Array is of -> Array of

File: pyuvdata/uvdata/uvdata.py [hunk 3]
```diff
@@ -5335,71 +5334,60 @@ def _apply_w_proj(self, new_w_vals, old_w_vals, select_mask=None):
...
                 inv_mask[select_mask] = False
-                select_mask = ~inv_mask
-                select_len = np.sum(select_mask)
+
+                if all(inv_mask):
+                    # If nothing is selected, then bail
+                    return
+
+                # If everything is selected by the mask, then no entries pop up in the
+                # inverse, in which case it's faster to use the Ellipsis
```
Comment (bhazelton) [inline pyuvdata/uvdata/uvdata.py:5348]: very helpful comment

**Review by bhazelton — COMMENTED** [review]
Looks good, just a couple of docstring comments.

File: pyuvdata/uvdata/uvdata.py (hunk 1 above)
Comment (kartographer) [reply pyuvdata/uvdata/uvdata.py]: Ah, good catch -- little too much blind copy/paste on my part. Edited with the above suggestion in 4190b5a0350e04050230984dcc531890e5c7d023

File: pyuvdata/uvdata/uvdata.py (hunk 2 above)
Comment (kartographer) [reply pyuvdata/uvdata/uvdata.py]: Fixed in 4190b5a0350e04050230984dcc531890e5c7d023

File: pyuvdata/uvdata/uvdata.py (hunk 3 above)
Comment (kartographer) [reply pyuvdata/uvdata/uvdata.py:5348]: Definitely a happy accidental discovery from the past few months..`Ellipsis`

**Review by bhazelton — APPROVED** [review]
Looks good!

**Pull request merged**